### PR TITLE
Update Puzzle value from chest

### PR DESCRIPTION
**Pull request by MiguelTrujillo26**
Refactor chest code.
update puzzle value each time the chest is opened.

### CODE REVIEW

File: web/src/api/InventoryAPI.js [hunk 1]
```diff
@@ -1,32 +1,33 @@
...
-            setError(error);
-            throw error;
-        } finally {
-            setLoading(false);
-        }
-    };
+    try {
+      const data = await patchData(url, {}, token);
+      setResponse(data);
+      return data;
```
Comment (andymotagl) [inline web/src/api/InventoryAPI.js:21]: If you are using a state, technically you won't need to return this

File: web/src/client/chest/Chest.js [hunk 2]
```diff
@@ -9,50 +9,71 @@ import { isSameDay } from '../../utils/dates.js';
...
 
-  const canOpenChest = (inventory) => {
-    if (!inventory.daily_obtained_at) return true;
-    return !isSameDay(inventory.daily_obtained_at);
-  };
+//TODO Define the logic for calculating the number of new puzzles
+const calculateValuePuzzle = (result) => {
+  const arraySize = result.data.length;
+  console.log('arraySize:', arraySize);
+  console.log('data:', result.data);
```
Comment (andymotagl) [inline web/src/client/chest/Chest.js:22]: remove these

File: web/src/client/chest/Chest.js [hunk 3]
```diff
@@ -9,50 +9,71 @@ import { isSameDay } from '../../utils/dates.js';
...
+  if (!result.data || arraySize === 0) {
+    return 0;
+  }
+  console.log('result.data[0].pieces[0]:', result.data[0].pieces.length);
```
Comment (andymotagl) [inline web/src/client/chest/Chest.js:26]: Same

File: web/src/client/chest/Chest.js [hunk 4]
```diff
@@ -9,50 +9,71 @@ import { isSameDay } from '../../utils/dates.js';
...
     if (isFetchingChest) return;
     setIsFetchingChest(true);
-    await generateDailyChest()
-      .then(() => {
+
+    try {
+      const result = await generateDailyChest();
+
+      if (result) {
+        console.log('Response updated:', result);
```
Comment (andymotagl) [inline web/src/client/chest/Chest.js:58]: remove

File: web/src/client/chest/Chest.js [hunk 5]
```diff
@@ -9,50 +9,71 @@ import { isSameDay } from '../../utils/dates.js';
...
+  }, [inventory.daily_obtained_at]);
+
   const getDailyChest = async () => {
     if (isFetchingChest) return;
     setIsFetchingChest(true);
-    await generateDailyChest()
-      .then(() => {
+
+    try {
+      const result = await generateDailyChest();
```
Comment (andymotagl) [inline web/src/client/chest/Chest.js:55]: This is in data state

File: web/src/client/chest/Chest.js [hunk 6]
```diff
@@ -9,50 +9,71 @@ import { isSameDay } from '../../utils/dates.js';
...
+        const value = calculateValuePuzzle(result);
+        setValuePuzzle((prevValue) => prevValue + value);
         setDisabled(true);
         setIsChestClosed(false);
-      })
-      .catch((e) => console.log(error))
-      .finally(() => setIsFetchingChest(false));
+      } else {
+        console.log('No se recibió un resultado válido', result);
+      }
```
Comment (andymotagl) [inline web/src/client/chest/Chest.js:65]: This is not needed

File: web/src/client/chest/Chest.js [hunk 7]
```diff
@@ -9,50 +9,71 @@ import { isSameDay } from '../../utils/dates.js';
...
+    } catch (e) {
+      console.log('Error al generar el cofre diario:', e);
```
Comment (andymotagl) [inline web/src/client/chest/Chest.js:67]: I guess instead of log the error we should show eventually something when this fails

**Review by fernando-arevalo-galilei — APPROVED** [review]

**Pull request merged**